cpp/modules/deck.gl/core/src/lib/attribute/attribute-descriptor.h

```diff
@@ -18,37 +18,32 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-#ifndef LUMAGL_WEBGPU_WEBGPU_COLUMN_H
-#define LUMAGL_WEBGPU_WEBGPU_COLUMN_H
+#ifndef DECKGL_CORE_ATTRIBUTE_ATTRIBUTE_DESCRIPTOR_H
+#define DECKGL_CORE_ATTRIBUTE_ATTRIBUTE_DESCRIPTOR_H
 
 #include <arrow/array.h>
+#include <arrow/table.h>
 #include <dawn/webgpu_cpp.h>
 
 #include <memory>
+#include <string>
+#include <utility>
 
 #include "luma.gl/webgpu/src/data/attribute-descriptor.h"
-#include "luma.gl/webgpu/src/webgpu-helpers.h"
 
-namespace lumagl {
+namespace deckgl {
 
-class WebGPUColumn {
+struct AttributeDescriptor : public lumagl::AttributeDescriptor {
  public:
-  WebGPUColumn(wgpu::Device device, const std::shared_ptr<AttributeDescriptor>& descriptor);
-  ~WebGPUColumn();
+  using AttributeBuilder = auto(const std::shared_ptr<arrow::Table>&) -> std::shared_ptr<arrow::Array>;
+  AttributeDescriptor(const std::string& name, const wgpu::VertexFormat format,
+                      std::function<AttributeBuilder> attributeBuilder)
+      : lumagl::AttributeDescriptor{name, format}, attributeBuilder{attributeBuilder} {}
```

Comment:
What if we just used arrow types here. We would not need to expose wgpu at all?

(You may need another field to record whether this is an instanced attribute. It can just be a boolean, but maps to https://gpuweb.github.io/gpuweb/#enumdef-gpuinputstepmode.)